src/ride/station.c

```diff
@@ -64,7 +64,10 @@ static void ride_update_station_blocksection(rct_ride *ride, int stationIndex)
 {
 	rct_map_element *mapElement;
 
+	assert(ride != NULL);
```

Comment:
Its not the functions fault if the caller passes null. The ride assert should not be here